components/GameForm.tsx

```diff
@@ -37,34 +46,53 @@ function GameForm(props) {
     reValidateMode: "onChange",
   });
 
+  if (gameId) {
+    const [games, setGames] = useState<any[]>([]);
+    const [hasLoadedGame, setHasLoadedGame] = useState(false);
+
+    useEffect(() => {
+      if (!hasLoadedGame) {
+        fetch(`api/games/`)
+          .then((res) => {
+            return res.json();
+          })
+          .then((data) => {
+            setGames(data.data);
+            setHasLoadedGame(true);
+          });
+      }
+    }, [gameId]);
+
+    const gameToEdit: GameFormValues = games.find((game) => game.id == gameId);
```

Comment:
Imagine there would be a thousand of games that would be returned from that endpoint, or if this endpoint would give you paginated games, so it's very likely that you wouldn't get game that you were looking for.

You should have a separate endpoint which would give you a single game -> `/api/games/[id]`